### PR TITLE
Refactor `do_partition` to reduce the statement count

### PR DESCRIPTION
**Pull request by jcjones**
ruff complains about routines with more than 50 statements, and `do_partition` had 52. #80 bumped that to 54, and we can just do a little refactor to make it cleaner.

Removes `max-statements = 52 # default is 50` from the `pyproject.toml` since we no longer need the override.

### CODE REVIEW

File: partitionmanager/cli.py [hunk 1]
```diff
@@ -273,6 +273,56 @@ def migrate_cmd(args):
 )
 MIGRATE_PARSER.set_defaults(func=migrate_cmd)
 
+def _partition_table(conf, log, table, metrics):
+    table_problems = pm_tap.get_table_compatibility_problems(conf.dbcmd, table)
+    if table_problems:
```
Comment (cclauss) [inline partitionmanager/cli.py:278]:
```suggestion
    if table_problems := pm_tap.get_table_compatibility_problems(conf.dbcmd, table):
```
https://pypi.org/project/auto-walrus

File: partitionmanager/cli.py [hunk 2]
```diff
@@ -273,6 +273,56 @@ def migrate_cmd(args):
...
+        log.error(f"Cannot proceed: {table} {table_problems}")
+        return None
+
+    map_data = pm_tap.get_partition_map(conf.dbcmd, table)
+
+    duration = conf.partition_period
+    if table.partition_period:
+        duration = table.partition_period
```
Comment (cclauss) [inline partitionmanager/cli.py:286]:
```suggestion
    duration = table.partition_period or conf.partition_period
```

**Review by cclauss — APPROVED** [review]
A few minor optimizations...

File: partitionmanager/cli.py [hunk 3]
```diff
@@ -306,56 +356,15 @@ def do_partition(conf):
...
-                continue
-
-            log.info(f"{table} running SQL: {composite_sql_command}")
-            time_start = datetime.utcnow()
-            output = conf.dbcmd.run(composite_sql_command)
-
-            all_results[table.name] = {"sql": composite_sql_command, "output": output}
-            log.info(f"{table} results: {output}")
+            results = _partition_table(conf, log, table, metrics)
+            if results:
```
Comment (cclauss) [inline partitionmanager/cli.py:361]:
```suggestion
            if results := _partition_table(conf, log, table, metrics):
```

**Review by cclauss — APPROVED** [review]
A few minor optimizations...

File: partitionmanager/cli.py (hunk 1 above)
Comment (jcjones) [reply partitionmanager/cli.py:278]: WALRUS!

**Review by ezekiel — APPROVED** [review]

**Pull request merged**